src/components/text-editor/prosemirror-adapter/plugins/image/inserter.ts

```
@@ -261,19 +283,253 @@
 
-    if (filteredSlice.content.childCount < slice.content.childCount) {
-        const { state, dispatch } = view;
-        const tr = state.tr.replaceSelection(filteredSlice);
-        dispatch(tr);
+/**
+ * Convert a URL or data URL to a Blob
+ *
+ * @param url - The URL or data URL to convert
+ * @returns A Promise that resolves to a Blob or null if conversion fails
+ */
+const urlToBlob = async (url: string): Promise<Blob | null> => {
```

ESLint

[error] 'urlToBlob' is assigned a value but never used. Allowed unused vars must match /^h$/u.

(@typescript-eslint/no-unused-vars)